Add job list page

superstar54 opened the pull request:
To help users easily search and manage their jobs, this PR adds a new job list page. This is the first step to fix #515 .

## Home page
On the home page, I added a new **Utils** section on the right side of the logo. Currently, there is only the `job list` item, but we can add more items in the future, e.g., 'plugin list' in #646 .

## Job list page
- show a table of the processes: pk, ctime, state, relaxed, properties
- **delete button and inspect button.**


## Demo

https://github.com/aiidalab/aiidalab-qe/assets/11457659/d3ddfce2-d32f-46fc-a7a1-0d1bcc98874e



## Future improvement
After this PR, open issues on
- [ ] add search bar

AndresOrtegaGuerrero commented:
@superstar54  Nice, I will test it and let you know, I was wondering. If an user has a WorkChain , maybe lets say a bands , but the structure as an input is coming from another WorkChain, and by deleting the first node it deletes automatically all the linked nodes. Is this avoided?

superstar54 commented:
> If an user has a WorkChain , maybe lets say a bands , but the structure as an input is coming from another WorkChain, and by deleting the first node it deletes automatically all the linked nodes. Is this avoided?

Good question. I will look into it and provide more information to the user when deleting nodes.

AndresOrtegaGuerrero commented:
@superstar54 I would suggest you add a section explaning that by deleting a node (QeWorkChain) all the associated and link nodes will be eliminated , as well as other calculations that might be linked with this node.

AndresOrtegaGuerrero commented on delete.ipynb:
```diff
@@ -0,0 +1,64 @@
+    "        print(f'Node {pk} deleted.')\n",
+    "\n",
+    "# Ask for confirmation\n",
+    "if input(f'Do you really want to delete node {pk}? [y/N] ').strip().lower() == 'y':\n",
```
Also display the number nodes associated, so the person know what nodes or workchains are being deleted

superstar54 replied:
Yes, I display the number of nodes to be deleted.

superstar54 commented:
> I really like the job list, is easy to search through jobs. I could be nice if there is a choice to add some filters like, filter by state : Finished[0] , or properties. I guess in this page the search app from edan could be useful.

Thanks, I added basic search filters. We can use @edan-bainglass 's query builder when it's ready in the future.

![Screenshot from 2024-04-08 17-04-14](https://github.com/aiidalab/aiidalab-qe/assets/11457659/e9c7a8a6-4603-4945-8cb8-d6bbda3ce814)

superstar54 commented:
from @cpignedoli and @giovannipizzi 
- Add label/description for search.
- allow edit the label/description.

superstar54 commented:
Not delete a job, but hide it.

superstar54 commented:
Hi @AndresOrtegaGuerrero , I made the changes as you suggested. Here is the screenshot of the page when user deleting a node. Could you please have another look?

![Screenshot from 2024-04-24 07-39-26](https://github.com/aiidalab/aiidalab-qe/assets/11457659/fe4d2030-704f-4735-833c-3db6f9f94bcc)

AndresOrtegaGuerrero commented on delete.ipynb:
```diff
@@ -0,0 +1,114 @@
+   "source": [
+    "# AiiDAlab QauntumESPRESSO App\n",
+    "\n",
+    "<font color=\"red\"><b>Caution!</b></font> Deleting this job will also remove <b>all associated nodes</b>, including any related calculations and their respective results. This action is <b>irreversible</b>.\n"
```
`<font color="red"><b>Caution!</b></font> Deleting this job will also remove <b>all associated nodes</b>, including any workflows that share a common input structure, as well as various calculations and their respective results. <b>This action is irreversible.</b>`

superstar54 replied:
Hi Andres, thanks for the comment. 
> including any workflows that share a common input structure
It will not delete the workflows that share the input structure. It only deletes its descendants. 
- If QEApp jobs A and B use the same input structure, if A is deleted, B will not.
- If QEApp job A has an output structure and job B uses it as input, then if A is deleted, B will also be deleted. I added a check here, if this happened, user can not delete the job.

AndresOrtegaGuerrero commented on delete.ipynb:
```diff
@@ -0,0 +1,114 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# AiiDAlab QauntumESPRESSO App\n",
```
AiiDAlab QuantumESPRESSO App

AndresOrtegaGuerrero commented on job_list.ipynb:
```diff
@@ -0,0 +1,57 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# AiiDAlab QauntumESPRESSO App\n",
```
# AiiDAlab QuantumESPRESSO App\n

AndresOrtegaGuerrero commented:
Hey Xing , it looks good to me , just check the few typos. I was wondering something, is it possible to check if the selected node to deete has other QeAppWorkChain associated ? and display this ? if not we can leave it for a future PR

superstar54 commented:
Hi @AndresOrtegaGuerrero , thanks for the review and suggestion. I have implemented the changes as you suggested.

> I was wondering something, is it possible to check if the selected node to deete has other QeAppWorkChain associated ? and display this ? if not we can leave it for a future PR


Yes, it's possible. I have added a check to see if there is a QEApp job using the outputs of the nodes that need to be deleted. If there is, the deletion will be blocked. Only after the user deletes the related QEApp job can the root one be deleted.

![Screenshot from 2024-04-24 12-06-28](https://github.com/aiidalab/aiidalab-qe/assets/11457659/ce08d24d-7f20-4383-8c7b-1392ec885c7e)

AndresOrtegaGuerrero reviewed (approved):
LGTM!
